src/app_engine/gcm_notify.py

```diff
@@ -44,94 +45,119 @@ def get_gcm_api_key():
   return GCM_API_KEY
 
 
-def send_gcm_messages(gcm_ids, message, collapse_key):
-  """Sends the message to each endpoint specified in gcm_ids.
+class GCMMessage(object):
+  def __init__(self, gcm_id, registration_id):
+    self.gcm_id = gcm_id
+    self.registration_id = registration_id
 
-  Makes a synchronous post to the GCM server with the appropriate payload
-  format.
+  def get_gcm_payload(self):
+    payload_dict = {
+        'data': self.get_data(),
+        'registration_ids': [self.gcm_id],
```

Comment:
Client doesn't know destination GCM ID. We probably don't want to stuff that in our GCM messages since it is sensitive.